scripts/01_Seurat_QC.Rmd

```diff
@@ -1,115 +1,110 @@
   log_file: "log" # log file
 ---
+  
+# Information about this notebook
+This script creates Seurat object from the aligned count data and performs QC based on Seurat functions. 
+
+For more information or updates on Seurat, please see: https://satijalab.org/seurat/articles/pbmc3k_tutorial.html
+
+
+# Usage
+
+To run the Rscript from the command line sequentially, use:
 
 ```
-Rscript -e "rmarkdown::render('01_Seurat_QC.Rmd', clean = TRUE,
+Rscript -e "rmarkdown::render('seurat_alignment_qc.Rmd', clean = TRUE,
             params=list(scooter_path = '/Users/chronia/CHOP/GitHub/scooter', 
-                        out_path='/Users/chronia/CHOP/projects/HOPE/HOPE-analysis/analyses/01_snRNA-seq/data/01_QC_objects/', 
+                        results_dir='/Users/chronia/CHOP/projects/HOPE/HOPE-analysis/analyses/01_snRNA-seq/data/01_QC_objects/', 
                         data_path='/Users/chronia/CHOP/projects/HOPE/HOPE-analysis/analyses/01_snRNA-seq/00-alignment/7316-2594_include_introns/outs/filtered_feature_bc_matrix/', 
                         sample_name='7316-2594_include_introns'))"
 ```
+# Directories and paths to file Inputs/Outputs
 
-```{r setup, include=FALSE}
-attach(params) 
-knitr::opts_chunk$set(echo = FALSE,
-                      warning = FALSE, 
-                      message = FALSE, 
-                      fig.path = file.path(out_path, paste0("Seurat_QC-", sample_name, "/")),
-                      fig.height = 10, 
-                      fig.width = 15,
-                      dev = c("png", "pdf"))
-```
-
-```{r load_libraries}
-library(future)
-library(cowplot)
-library(devtools)
-library(forcats)
-library(GGally)
-library(stringr)
-library(ggpmisc)
-library(ggrepel)
-load_all(scooter_path)
-theme_set(theme_bw())
-
-# # evaluate Seurat R expressions asynchronously when possible (such as ScaleData) using future package
-plan("multisession", workers = 4)
-# # increase the limit of the data to be shuttled between the processes from default 500MB to 50GB
-options(future.globals.maxSize = 30 * 1024 ^ 3)
-```
+```{r set-dir-and-file-names, echo=TRUE}
+# File path to data with the filtered_feature_bc_matrix
+data_path <- file.path("/Users/chronia/CHOP/GitHub/kf-dev-single-cell-rna/scripts/example/data/pbmc_1k_v3/outs/filtered_feature_bc_matrix")
+if (dir.exists(data_path)) {
+    cat(paste(c("", sprintf("`%s`", list.files(paste(data_path, "/outs"), full.names = TRUE))),
+              collapse = "\n- "))
+} else { # hd5 file
+    cat(paste("Input file:", sprintf("`%s`", data_path), ".\n"))
+}
+cat("\n\n")
 
-## Directories and File Inputs/Outputs
-```{r set-dir-and-file-names}
 # File path to results directory
-out_path <-
-  file.path(out_path, paste0("Seurat_QC-", sample_name, "/"))
-if (!dir.exists(out_path)) {
-  dir.create(out_path)
+results_dir <-
+  file.path("/Users/chronia/CHOP/GitHub/kf-dev-single-cell-rna/scripts/example/results", paste0("Seurat-QC-", sample_name, "/"))
```

Comment:
Hardcoded path. Please revert to the old approach where this can be handed in as a param.